sprints/sustainability/utils.py

```diff
@@ -40,17 +40,26 @@ def split_accounts_into_categories(accounts: List[Account]) -> Dict[str, List[Ac
     return result
 
 
-def generate_month_range(start: str, end: str) -> Generator[Tuple[datetime, datetime], None, None]:
+def generate_month_range(start: str, end: str) -> Generator[Tuple[datetime.date, datetime.date], None, None]:
     """Generates months between `start` and `end` dates."""
     start_date = parse(start)
     end_date = parse(end)
 
     current_date = start_date.replace(day=1)  # First day of each month
-    while current_date < end_date:
-        last_day_of_month = calendar.monthrange(current_date.year, current_date.month)[1]
-        last_date_of_month = current_date.replace(day=last_day_of_month)
+    while current_date <= end_date:
+        last_date_of_month = current_date + relativedelta(day=31)
```

Comment:
Months don't always have 31 days.
Won't this wrongly take into account a few days from the following month if the month has 30 or less days?

Reply:
`relativedelta` does `max(31, last_day_of_the_month)` :)

It actually implements more handy things, like "Last friday in this month." [which you can grep in these examples](https://dateutil.readthedocs.io/en/stable/examples.html).

Reply:
Woah, nice! That's super handy.

Reply:
One more subtle thing here about singulars and plurals for `relativedelta` - it might be handy in other parts of this review. Singulars **replace** the part within a range, while plurals **calculate the actual delta**:
```
year, month, day, hour, minute, second, microsecond:
    Absolute information (argument is singular); adding or subtracting a
    relativedelta with absolute information does not perform an arithmetic
    operation, but rather REPLACES the corresponding value in the
    original datetime with the value(s) in relativedelta.

years, months, weeks, days, hours, minutes, seconds, microseconds:
    Relative information, may be negative (argument is plural); adding
    or subtracting a relativedelta with relative information performs
    the corresponding arithmetic operation on the original datetime value
    with the information in the relativedelta.
```